stubs/pygit2/pygit2/blob.pyi

```diff
@@ -0,0 +1,25 @@
+import io
+import types
+from _typeshed import WriteableBuffer
+from contextlib import AbstractContextManager
+
+from ._pygit2 import Blob, Oid
+from .enums import BlobFilter
+
+class _BlobIO(io.RawIOBase):
+    def __init__(self, blob: Blob, as_path: str | None = None, flags: BlobFilter = ..., commit_id: Oid | None = None) -> None: ...
+    def __exit__(
+        self, exc_type: type[BaseException] | None, exc_value: BaseException | None, traceback: types.TracebackType | None
+    ) -> None: ...
+    def isatty() -> bool: ...  # type: ignore[misc]
```

Comment:
Seems like a runtime bug

Reply:
Yes this is an actual bug in the upstream library, and fixing it would be another separate issue to file on their repo.